commerce/domain/orchestrators/classes/CartCalculateSample.cls

```diff
@@ -103,4 +104,9 @@ global class CartCalculateSample extends CartExtension.CartCalculate {
 
         return false;
     }
+
+    private Boolean isCouponAppliedInCheckout(CartExtension.BuyerActions buyerActions, CartExtension.Cart cart) {
+        return (cart.getStatus() == CartExtension.CartStatusEnum.CHECKOUT || buyerActions.isCheckoutStarted()) &&
```

Comment:
I thought about it more. this implies we can have more than one buyer action at a time (isCheckoutStarted and isCouponChanged). @jbisaSF is that possible?

Reply:
@smahbubani99 for the unit test though, can we create a `Cart` that has `CartStatusEnum` set to `CHECKOUT` ? This way, we don't need to rely on `buyerActions.isCheckoutStarted()` ?

Reply:
@jbisaSF we do reject the coupon request if async processing is true.
@smahbubani99 my concern with mixing buyerActions.isCheckoutStarted() && isCouponAppliedInCheckout is that this can cause confusion to partner developers.

Reply:
@jbisaSF I don't believe we can w/ our current setup since we're not mocking CartExtension.Cart

Reply:
I could modify CartTestUtil though and get approval from the cart team

Reply:
this should be removed, right?
`|| buyerActions.isCheckoutStarted()`

we only need this:
`return (cart.getStatus() == CartExtension.CartStatusEnum.CHECKOUT) && buyerActions.isCouponChanged();`